lib/vector-buffers/src/variants/disk_v2/tests/invariants.rs

```diff
@@ -249,10 +255,7 @@ async fn writer_stops_when_hitting_file_that_reader_is_still_on() {
 
             // Now we should be consuming all data files, and our next write should block trying to
             // open the "first" data file until we do a read.
-            let mut blocked_write = spawn(async {
-                let record = SizedRecord(record_size);
-                writer.write_record(record).await
-            });
+            let mut blocked_write = spawn(async { writer.write_record(record).await });
```

Comment:
Can this be done without the `async` wrapper? It looks like `write_record` is just a future…
```suggestion
            let mut blocked_write = spawn(writer.write_record(record));
```

Reply:
Yep, I agree with your assessment.